packages/eventual-send/src/postponed.js

```diff
@@ -0,0 +1,48 @@
+// @ts-check
+/// <reference types="ses" />
+
+/**
+ * @template T
+ * @typedef {import('.').EHandler<T>} EHandler
+ */
+
+/**
+ * @param {import('.').HandledPromiseConstructor} HandledPromise
+ * @returns {[Required<EHandler<any>>, () => void]}
+ */
+export const makePostponedHandler = HandledPromise => {
+  // Create a simple postponedHandler that just postpones until the
+  // fulfilledHandler is set.
+  let donePostponing;
+  const interlockP = new Promise(resolve => {
+    donePostponing = () => resolve(undefined);
+  });
+
+  const makePostponedOperation = postponedOperation => {
```

Comment:
A caught an irregularity because of the declared types, even though for some reason TS doesn't see a problem.

`makePostponedOperation(str)(x, ...args)` always returns a promise. But index.d.ts says the *Only methods return `void`. Should we allow these methods to return promises, or should we in fact require them to return void?

Reply:
> `makePostponedOperation(str)(x, ...args)` always returns a promise. But index.d.ts says the *Only methods return `void`. Should we allow these methods to return promises, or should we in fact require them to return void?

I think this might be a misunderstanding.  TS `void` does not mean `undefined`, it means "I don't care what it is, and make sure nobody depends on this value".

It's fine for handlers to return something from their `*Only` methods, we just don't use that value anywhere.

Reply:
If it isn't way too much trouble, I think it would actually be better if *Only methods always returned `undefined`. I think it is especially bad for them to return promises, as code may take the resolution of that promise to mean something it does not. Returning `undefined` will help people catch some bugs earlier.